internal/net/phttp/warning_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package phttp
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"go.pinniped.dev/internal/constable"
+	"go.pinniped.dev/internal/httputil/roundtripper"
+)
+
+func Test_warningWrapper(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name       string
+		resp       *http.Response
+		wantCodes  []int
+		wantAgents []string
+		wantTexts  []string
+	}{
+		{
+			name:       "nil resp",
+			resp:       nil,
+			wantCodes:  nil,
+			wantAgents: nil,
+			wantTexts:  nil,
+		},
+		{
+			name:       "no warning",
+			resp:       testResp(t, http.Header{"moon": {"white"}}), //nolint:bodyclose
+			wantCodes:  nil,
+			wantAgents: nil,
+			wantTexts:  nil,
+		},
+		{
+			name:       "malformed warning",
+			resp:       testResp(t, http.Header{"Warning": {"wee"}}), //nolint:bodyclose
+			wantCodes:  nil,
+			wantAgents: nil,
+			wantTexts:  nil,
+		},
+		{
+			name:       "partial malformed warning",
```

Comment:
Add a test for partial malformed warning in the other order (e.g. malformed first, then good warning)

Reply:
Done.